notebook/notebookapp.py

```diff
@@ -825,10 +826,28 @@ def init_configurables(self):
             kernel_manager=self.kernel_manager,
             contents_manager=self.contents_manager,
         )
-        self.config_manager = self.config_manager_class(
-            parent=self,
-            log=self.log
-        )
+        if self.config_manager_class is ConfigManager:
+            # We know our ConfigManager ignore `config_dir`, 
+            # and will warn/raise a PendingDeprecation warning
+            # if given, so do not pass it (not to annoy user and show the
+            # warning on default install. .
+            self.config_manager = self.config_manager_class(
+                parent=self,
+                log=self.log,
+            )
+        else :
+            # In case it is a custom ConfigManager,  
+            # for potentially not breaking future backward 
+            # compatibility, will still pass it. 
+            if issubclass(self.config_manager_class, ConfigManager):
+                warnings.warn('Object `%s` will be passed a `config_dir` kwarg, which might be removed'
+                              'in future Notebook versions. Please make sure to handle this case.' % (self.config_manager_class),)
+            self.config_manager = self.config_manager_class(
+                parent=self,
+                log=self.log,
+                config_dir=os.path.join(self.config_dir, 'nbconfig'),
```

Comment:
This is the whole thing... we **don't** need the `config_dir` from the notebookapp for the custom ConfigManager... this is my simple implementation of a CustomConfig manager: https://github.com/Anaconda-Server/nb_config_manager/blob/master/nb_config_manager/nb_config_manager.py

Since the `config_dir` is "calculated" inside the ConfigManager class (by the `_config_dir_default`), my subclass take that info and make further modifications from there... so, we are happy with just the if branch above, where the `config_dir` is not harcoded at the notebook level... maybe I am not clear enough in my explanations... sorry about that.

Check my comment in https://github.com/jupyter/notebook/issues/893#issuecomment-167099432, with that diff you keep the things as now (I mean, not config option at the ConfigManager and the default config pointing to the user space), but you "hardcode" the config_dir at the ConfigManager level, so we can then subclass and just setup the config_dir to our desires without touching the notebookapp at all...